### PR TITLE
Add missing tests `ops/nn.py`

### PR DESCRIPTION
**Pull request by Faisal-Alsrheed**
I have added some tests for untested lines in `ops/nn.py` 

and

I will investigate this failed test later (another PR) 

TODO: Currently, when using the `one_hot` function with an invalid `axis` parameter, the error is directly thrown by the backends.
```
def test_one_hot_with_invalid_axis_raises_value_error(self):
        x = np.array([[0, 1], [1, 2]])
        num_classes = 3
        invalid_axis = 5
        expected_error_message = (
            f"axis must be -1 or between \\[0, {len(x.shape)}\\), but "
            f"received {invalid_axis}."
        )

        with self.assertRaisesRegex(ValueError, expected_error_message):
            knn.one_hot(x, num_classes=num_classes, axis=invalid_axis)
```

### CODE REVIEW

**Comment by Faisal-Alsrheed** [comment]
Currently, when using the `one_hot` function with an invalid `axis` parameter, the error is directly thrown by the backend (e.g., TensorFlow's `InvalidArgumentError`), which can lead to inconsistencies between different backends. 

@fchollet Any idea how to fix this?

**Comment by fchollet** [comment]
You could add a new error directly into `keras.ops.one_hot`?

**Comment by Faisal-Alsrheed** [comment]
@fchollet  
Done

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
